src/dbnode/storage/cleanup.go

```diff
@@ -146,27 +148,32 @@ func (m *cleanupManager) WarmFlushCleanup(t time.Time) error {
 	multiErr := xerrors.NewMultiError()
 	if err := m.cleanupExpiredIndexFiles(t, namespaces); err != nil {
 		multiErr = multiErr.Add(fmt.Errorf(
-			"encountered errors when cleaning up index files for %v: %v", t, err))
+			"encountered errors when cleaning up index files for %v: %w", t, err))
+	}
+
+	if err := m.cleanupCorruptedIndexFiles(namespaces); err != nil {
```

Comment:
This makes me think why does is the cleanup process called from `WarmFlushCleanup`. It would clean up cold flush index filesets as well, right? Also, from what I can see it covers all namespaces, including computed namespaces which perform no warm/cold flushing. Which makes the code somewhat confusing. Perhaps some refactoring/naming improvement could be useful here.

Reply:
It's not just this fn, all of the index file cleanup fns are the same. We only cleanup data files in the cold flush loop and index files in the warm flush loop. IIRC we only have cold flush cleanup because it was not safe to cleanup data files in the warm flush loop (can't remember the exact reason). Otherwise, we would've done all cleanup in the warm flush loop.